yaml/semgrep/key-indentation.yaml

```diff
@@ -0,0 +1,26 @@
+rules:
+  - id: yaml-key-indentation-check 
+    message: >-
+      It looks like you have an YAML indentation issue -- instead of writing `$KEY`, 
+      put a space between the hyphen and what comes after! Otherwise, it reads 
+      as a single string. 
+    languages:
+      - yaml
+    severity: WARNING
+    metadata:
+      references:
+        - https://semgrep.dev/docs/contributing/contributing-to-semgrep-rules-repository
+      category: correctness
+      technology:
+        - semgrep
+    pattern-either:
+      - patterns:
+        - pattern-inside: |
+            rules: ...
+        - pattern: |
+            $KEY: >-
+              $VALUE
+        - focus-metavariable: $KEY
+        - metavariable-regex:
+            metavariable: $KEY
+            regex: "^-(.*)$"
```

Comment:
Wouldn't this detect on anything after the dash, including spaces and stuff? Did you mean to do `"^-(\w)*$"`?

Reply:
@enncoded ooh, i think you're right! i was thinking it wouldn't matter, since the range of `$KEY` should be just that of the entire key to the pair, but it doesn't hurt to change -- fixed!

Reply:
The regex as you changed it to is invalid, but doing `regex: ^-(\w*)$` works. Also update the test to include an `#ok: ruleid` check. Feel free to copy/paste from https://semgrep.dev/playground/s/50N0